### PR TITLE
major: LANZ-2678 refactor gh action. added scope validation

### PR DESCRIPTION
**Pull request by cangulomascarell**

### CODE REVIEW

**Comment by cangulomascarell** [comment]
# Next features
## Scopes validation
* now the ensure-conventional-commits-gh-action can accept scopes as input parameters. It should be defined as a coma separated list. Commits with scope will be validated against that list. If the scope provided is in the list, commit OK, if not, commit NOK.
* If a commit doesn't have a scope, the other format validations will still be executed
## Match files modified against scope
* Files modified per commit can be validated in the next scenario: 
1. Set the app names as the scope list. 
2. .Set the `validate-scopes-against-apps` to true
3. If your apps are NOT in the root folder, provide their partial path in `apps-folder`. example: `src/`

### requirements
* Apps names must match the folder names
* All apps must be under the same directory

File: main.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+let customCCTypes = []
+if (customCCFilePath !== 'not provided') {
+  useCustomCC = true;
+  const rawData = fs.readFileSync(customCCFilePath);
+  customCCTypes = JSON.parse(rawData).map(x => x.commitType)
+  logger.logKeyValuePair("custom-cc-types", customCCTypes);
+}
+
+logger.logKeyValuePair("using-custom-conventional-commits", useCustomCC);
+const commitTypesAccepted = useCustomCC ? customCCTypes : defaultCCTypes
```
Comment (lucky-edu) [inline main.js:48]: are default and custom CC exclusive?

File: modules/commit-parser.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    if (!matchResult) {
+        logger.logValidationError(commit, "Commit does not follow basic pattern, check expected values next.", CONSTANTS.COMMIT_PATTERN_EXAMPLES)
+        process.exit(1);
+    }
+
+    let { type, scopes, breaking, body } = matchResult.groups
+
+    scopes = scopes ? scopes.split(",") : []
+    scopesProvided = scopes.length === 0 ? false : true
```
Comment (lucky-edu) [inline modules/commit-parser.js]: you can remove the '? false : true'

File: modules/commit-parser.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    isBreaking = breaking === "!" ? true : false
```
Comment (lucky-edu) [inline modules/commit-parser.js]: you can remove the '? false : true'

File: modules/git.js [hunk 4]
```diff
@@ -16,14 +16,16 @@ const prettyFormat = [
   "",
 ];
 
-const getCommitsInsidePullRequest = (destinationBranchName, branchName) => {
-  let mergeBaseCommit = child
-    .execSync(`git merge-base origin/${destinationBranchName} ${branchName}`)
-    .toString("utf-8")
-    .split("\n")[0];
+const getMergeBaseCommit = (targetBranchName, sourceBranch) => child
```
Comment (lucky-edu) [inline modules/git.js]: change targetBranchName to targetBranch

File: modules/commit-parser.js (hunk 2 above)
Comment (cangulomascarell) [reply modules/commit-parser.js]: ep, true. thanks

File: modules/commit-parser.js (hunk 3 above)
Comment (cangulomascarell) [reply modules/commit-parser.js]: yep, thanks

File: modules/git.js (hunk 4 above)
Comment (cangulomascarell) [reply modules/git.js]: thanks

File: main.js (hunk 1 above)
Comment (cangulomascarell) [reply main.js:48]: yes, you can only use either custom one or default ones

**Review by lucky-edu — APPROVED** [review]

**Pull request merged**